flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.formats.csv;
+
+import org.apache.flink.formats.common.TimeFormats;
+import org.apache.flink.table.planner.runtime.utils.TestData;
+import org.apache.flink.table.planner.utils.JavaScalaConversionUtil;
+import org.apache.flink.table.planner.utils.JsonPlanTestBase;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.table.utils.DateTimeUtils.toLocalDateTime;
+
+/** Tests for the CSV file format. */
+public class TableCsvFormatITCase extends JsonPlanTestBase {
+
+    @Test
```

Comment:
Same as for the other test class can you use junit 5 and assertJ, please.

Reply:
This test is largely based on utilities from the `JsonPlanTestBase` and `AbstractTestBase` base classes. I would prefer to leave the migration of those utility classes for table API to someone who works on Table/SQL. What do you think?

Reply:
We will probably take over all testing utils for formats. The right way would be to convert the `JsonPlanTestBase` into a junit extension and get rid of the inheritance.

I can understand if you do not want to do it as part of this PR but eventually with the next format we have to do it :)

Reply:
Yeah,  `AbstractTestBase` which is JUnit4-based is used in 400 classes, I propose to address it in a separate effort :)

Reply:
I mean you do not have to touch `AbstractTestBase` it does not offer much besides a MiniCluster which you can also get with the existing extension. The only effort would be to rework `JsonPlanTestBase`.

Reply:
`JsonPlanTestBase` is also used in 28 tests :sweat_smile: . I created a separate `TableFormatTestBase` based on JUnit5 and migrated my test to it.

Reply:
Please let's keep `JsonPlanTestBase` and don't add a new test base. The test base you added is based on legacy as well (`TestValuesTableFactory`), and it's doing assertions based on strings while we want to use the new `TableAssertions`. So your new base will immediately become just another legacy to clean up later :smile:. 

We're planning to do proper rework of our bases and push them in the new `flink-table-test-utils`, and these needs to be independent of table planner, so connector and formats stop depending on table planner test classpath. If you wanna create a new test base, feel free to start from this point: push it to `flink-table-test-utils` and use the new `TableAssertions`. But I'm pretty sure it's quite some work, so no need to do it in this PR.

Reply:
Reverted back to the `JsonPlanTestBase`.